lib/RefBackend/RefBackend.cpp

```diff
@@ -219,74 +220,189 @@ mlir::torch::RefBackend::createMungeCallingConventionsPass() {
 }
 
 //===----------------------------------------------------------------------===//
-// InsertRngGlobals
+// MLProgramBufferize
 //===----------------------------------------------------------------------===//
 
-static constexpr StringRef getSeedGobalVarName() { return "global_seed"; }
+static MemRefType
+getMemrefTypeForGlobalOp(Type type,
+                         llvm::Optional<RankedTensorType> &tensorType) {
+  MemRefType memrefType;
+  if (type.isIntOrFloat()) {
```

Comment:
please remove support for non-tensor ml_program.global's. Those code paths are unused in our flow and so are not tested end-to-end.